WalletWasabi.Documentation/Guides/DeterministicBuildGuide.md

```diff
@@ -1,67 +1,79 @@
 
-In order to reproduce Wasabi's builds you need Git, Windows 10 and the version of .NET Core SDK that was the most recent in the time of building the release.
+This guide describes how to reproduce Wasabi's builds. If you get stuck with these instructions, take a look at [how to build Wasabi from source code](https://github.com/zkSNACKs/WalletWasabi#build-from-source-code).
 
-# 2. Reproduce Builds
+**Warning:** Reproducible builds were introduced in [1.1.3 release](https://github.com/zkSNACKs/WalletWasabi/releases/tag/v1.1.3), you cannot use these instructions for older versions!
+
+## 1. Assert correct environment
+
+In order to reproduce Wasabi's builds, you need [git](https://git-scm.com/) package, Windows 10, and the version of .NET Core SDK that was used by the Wasabi team to produce given Wasabi Wallet release. The latest version of .NET Core SDK is always used, unless specified otherwise in the release notes of Wasabi Wallet. You can get it here https://dotnet.microsoft.com/download/dotnet-core.
+
+## 2. Reproduce builds
+
+You can see the list of Wasabi releases here: https://github.com/zkSNACKs/WalletWasabi/releases. Please note that each release has a version and a git hash assigned. The git hash is useful in the following instructions:
 
 ```sh
 git clone https://github.com/zkSNACKs/WalletWasabi.git
-git checkout {hash of the release} # This works from 1.1.3 release, https://github.com/zkSNACKs/WalletWasabi/releases
-cd WalletWasabi/WalletWasabi.Packager/
+cd WalletWasabi
+git checkout <SHA-1-hash-of-the-release>
+cd WalletWasabi.Packager
 dotnet restore
 dotnet build
 dotnet run -- --onlybinaries
 ```
 
-This will build our binaries for Windows, OSX and Linux from source code and open them in a file explorer for you.
+These commands will produce Wasabi's binaries for Windows, macOS and Linux from source code. Also, for your convenience, a new file explorer window will navigate you to the binaries location - i.e. `WalletWasabi\\WalletWasabi.Gui\\bin\\dist`.
 
 ![](https://i.imgur.com/8XAQzz4.png)
 
-# 3. Verify Builds
+## 3. Verify builds
 
-You can compare our binaries with the downloads we have on the website: https://wasabiwallet.io/
-In order to end-to-end verify all the downloaded packages you need a Windows, a Linux, and an OSX machine.
+Now, we will attempt to verify the binaries you have just compiled with the officially distributed binaries on https://wasabiwallet.io website. Please download those packages from the website, you should see the following files in your File Explorer:
 
 ![](https://i.imgur.com/aI9Kx0c.png)
 
-## Windows
+Please note that to completely verify Wasabi packages for all supported platforms, you actually need machines with Windows, Linux and macOS operating systems. If you don't have so many physical machines, you can use any virtualization sofware packages like [VirtualBox](https://www.virtualbox.org/), [VMWare](https://www.vmware.com/), etc.
```

Comment:
Maybe I should add: `or you can use WSL as described below`?

Reply:
NACK
I am using windows to release. To be able to reproduce the build one has to use windows (and WSL) to be able to do that. .NET builds differs if you compile on different OS even if the deterministic flag was set to true at dotnet publish command.

Reply:
@molnard `In order to end-to-end verify all the downloaded packages you need a Windows, a Linux, and an OSX machine.` - this sentence confused me. I'll modify this.

Reply:
@molnard I have dropped the sentence completely.